tools/payloads/ysoserial/findYsoserialOffsets.rb

```diff
@@ -0,0 +1,179 @@
+g#!/usr/bin/env ruby
+# encoding: binary
+
+#TODO: Remove previous line?
+
+# ASSUMPTION: A payload will not exceed 255 bytes  (TODO: That's a pretty terrible assumption.)
+# ASSUMPTION: A payload will have only one length offset and one payload string offset (this has held true on 13 of 13 tested payloads)
+
+require 'diff-lcs'
+require 'json'
+require 'base64'
+require 'open3'
+
+YSOSERIAL_RANDOMIZED_HEADER = "ysoserial\/Pwner"
+PAYLOAD_TEST_MAX_LENGTH = 5
+
+def generatePayload(payloadName,searchStringLength)
+  program = "ysoserial-original.jar"
+  #STDERR.puts " Generating #{payloadName} with length #{searchStringLength} using #{program}"
+
+  # Generate a string of specified length and embed it into an ASCII-encoded ysoserial payload
+  searchString = 'A'*searchStringLength
+  #STDERR.puts "  Calling java -jar #{program} #{payloadName} '#{searchString}'"
+  stdout, stderr, status = Open3.capture3("java -jar #{program} #{payloadName} '#{searchString}'")
```

Comment:
This is probably safer:

```suggestion
  stdout, stderr, status = Open3.capture3('java', '-jar', program.to_s, payloadName.to_s, searchString.to_s)
```